apps/aggiemap/src/app/modules/services/trip-planner/trip-planner.service.ts

```diff
@@ -444,20 +448,32 @@ export class TripPlannerService implements OnDestroy {
     // additional methods when both streams complete.
     zip(moduleProvider.require(['RouteTask', 'RouteParameters', 'FeatureSet', 'Graphic'], true), mapService.store)
       .pipe(takeUntil(this.$destroy))
-      .subscribe((results: [any, MapServiceInstance]) => {
-        this._Modules.TripTask = results[0].RouteTask;
-        this._Modules.TripParameters = results[0].RouteParameters;
-        this._Modules.FeatureSet = results[0].FeatureSet;
-        this._Modules.Graphic = results[0].Graphic;
+      .subscribe(
+        (
+          results: [
+            {
+              RouteTask: esri.RouteTaskConstructor;
+              RouteParameters: esri.RouteParametersConstructor;
+              FeatureSet: esri.FeatureSetConstructor;
+              Graphic: esri.GraphicConstructor;
+            },
```

Comment:
Add `RouteTaskConstructor` and `RouteParametersConstructor` to `TripPlannerModules` interface

Reply:
Once that's done, can probably be simplified using the `pluck` operator. 

```
      .pipe(pluck('results'), takeUntil(this.$destroy))
      .subscribe(
        ([
            modules,
            instance
          
         ]: [TripPlannerModules, MapServiceInstance]) => {
          this._Modules.TripTask = modules.RouteTask;
          this._Modules.TripParameters = modules.RouteParameters;
          this._Modules.FeatureSet = modules.FeatureSet;
          this._Modules.Graphic = modules.Graphic;

          // Locally store instance of map and view, allowing direct map and view manipulation
          this._map = instance.map;
          this._view = instance.view;

          this.loadTripFromURL();

          this.initializeHandlers();
        }
      );
```